lxd/auth/util.go

```diff
@@ -42,7 +42,23 @@ func IsRootUserFromCtx(ctx context.Context) (bool, error) {
 		return true, nil
 	}
 
-	return false, nil
+	id, err := GetIdentityFromCtx(ctx, identityCache)
+	if err != nil {
```

Comment:
Should we check for an explicit "not found" error rather than any error, in case some internal problem occurs retrieving the identity from the cache that may then trigger this request to become treated as admin?

Reply:
Oh you already thought of this below :)